File: spec/factories/items.rb
FactoryBot.define do
  factory :item do
    name { Faker::Commerce.product_name }
    description { Faker::Lorem.sentence }
    price { Faker::Commerce::price.floor }
    image do
      Rack::Test::UploadedFile.new(
        Rails.root.join('spec/fixtures/platanus.png'),
        'image/png'
      )
    end
  end
end

File: app/javascript/api/items.ts
import type { Review } from "./reviews";

export interface Item {
  id: number;
  name: string;
  description: string;
  price: number;
  image: Record<string, Record<'url', string>>;
  reviewsCount: number;
  reviewsAverage: number;
  reviews: Review[]
}
